### PR TITLE
Platui 54

### PR DESCRIPTION
**Pull request by ejayaraman**
PLATUI-54: 
ZAP tests are included in the using the ui-test-template. Creating a new test repo from the template will have 
 -  a sample zap-automation config in application. conf
 - a separate UI test tagged as ZapTests
 - a shell script to execute the ZapTest

README updated and tests added.

### CODE REVIEW

File: src/main/g8/README.md [hunk 1]
```diff
@@ -27,10 +27,42 @@ To run the tests against an environment set the corresponding `host` environment
...
  [webdriver-factory](https://github.com/hmrc/webdriver-factory#executing-against-a-test-environment) library. 
  
-For example, to execute the `run_tests.sh` script against QA  environment using chrome remote-webdriver
+For example, to execute the `run_tests.sh` script against QA  environment using Chrome remote-webdriver
      
     ./run_tests.sh qa remote-chrome
 
+## Running ZAP tests
+
+ZAP tests are implemented using the HMRC [zap-automation](https://github.com/hmrc/zap-automation) library.
```
Comment (dannyprok) [inline src/main/g8/README.md]: change to: 

ZAP tests can be automated using the HMRC [zap-automation](https://github.com/hmrc/zap-automation) library.  It is not mandatory to do so and should not be considered a substitute for manual exploratory testing using OWASP ZAP.

File: src/main/g8/build.sbt [hunk 2]
```diff
@@ -7,6 +9,7 @@ scalaVersion := "2.11.11"
 scalacOptions ++= Seq("-unchecked", "-deprecation", "-feature")
 
 resolvers += "hmrc-releases" at "https://artefacts.tax.service.gov.uk/artifactory/hmrc-releases/"
+resolvers += Resolver.bintrayRepo("hmrc", "releases")
```
Comment (dannyprok) [inline src/main/g8/build.sbt]: do we need:
`resolvers += "hmrc-releases" at ....`
if we include:
`resolvers += Resolver.bintrayRepo("hmrc", "releases")`

in build.sbt?

File: src/main/g8/README.md (hunk 1 above)
Comment (TomDudley92) [reply src/main/g8/README.md]: I agree with this change here

File: src/main/g8/run_zap_tests.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+elif [ "\$BROWSER" = "firefox" ]; then
+    DRIVER="-Dwebdriver.gecko.driver=/usr/local/bin/geckodriver"
+fi
+
+# NOTE: It is not required to proxy every journey test via ZAP. The intention of proxying a test through ZAP is to expose all the
+# relevant pages of an application to ZAP. So tagging a subset of the journey tests or creating a
+# single ZAP focused journey test is sufficient.
+
+$if(cucumber.truthy)$
+sbt -Dbrowser=\$BROWSER -Denvironment=\$ENV \$DRIVER -Dzap.proxy=true 'test-only uk.gov.hmrc.test.ui.cucumber.runner.ZapRunner'
```
Comment (TomDudley92) [inline src/main/g8/run_zap_tests.sh]: Shouldn't the ' be " here?

File: src/main/g8/src/test/resources/$if(cucumber.truthy)$features$endif$/$if(cucumber.truthy)$Example.feature$endif$ [hunk 4]
```diff
@@ -5,4 +5,10 @@ Feature: Example Feature file using Cucumber
   Scenario: A logged in user is able to access payment details page
     Given a user logs in to access payments page
     When the user chooses to pay VAT tax
-    Then payment details page is displayed
\ No newline at end of file
+    Then payment details page is displayed
+
+  @ZAP
```
Comment (TomDudley92) [inline src/main/g8/src/test/resources/$if(cucumber.truthy)$features$endif$/$if(cucumber.truthy)$Example.feature$endif$]: Could this tag not be added to the Scenario above?

File: src/main/g8/src/test/scala/uk/gov/hmrc/test/ui/$if(!cucumber.truthy)$specs$endif$/$if(!cucumber.truthy)$ExampleSpec.scala$endif$ [hunk 5]
```diff
@@ -31,5 +32,28 @@ class ExampleSpec extends BaseSpec {
       Then("Choose a way to pay page is displayed")
       tagName("h1").element.text shouldBe "Choose a way to pay"
     }
+
+    //Test tagged for ZAP
```
Comment (TomDudley92) [inline src/main/g8/src/test/scala/uk/gov/hmrc/test/ui/$if(!cucumber.truthy)$specs$endif$/$if(!cucumber.truthy)$ExampleSpec.scala$endif$]: Again does this need to be repeated? A comment could be added showing that this is how Zap is added?

File: src/main/g8/run_zap_tests.sh (hunk 3 above)
Comment (dannyprok) [reply src/main/g8/run_zap_tests.sh]: 👍

File: src/main/g8/src/test/scala/uk/gov/hmrc/test/ui/$if(!cucumber.truthy)$specs$endif$/$if(!cucumber.truthy)$ExampleSpec.scala$endif$ (hunk 5 above)
Comment (dannyprok) [reply src/main/g8/src/test/scala/uk/gov/hmrc/test/ui/$if(!cucumber.truthy)$specs$endif$/$if(!cucumber.truthy)$ExampleSpec.scala$endif$]: tag moved, scenario deleted.

File: src/main/g8/src/test/resources/$if(cucumber.truthy)$features$endif$/$if(cucumber.truthy)$Example.feature$endif$ (hunk 4 above)
Comment (dannyprok) [reply src/main/g8/src/test/resources/$if(cucumber.truthy)$features$endif$/$if(cucumber.truthy)$Example.feature$endif$]: tag moved, scenario deleted

**Pull request merged**